### PR TITLE
feat: added docker-compose for running postgres

### LINKED ISSUE #35: [FEATURE] docker setup for local development
### Description

Currently, To run the project locally we have to run the postgress database seperately.
We can use docker to automate this process so that it's easier for people to setup the local environment.

I would like to work on this issue.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sital002**
## Fixes Issue
closes #35 

## Changes proposed
Added a docker compose file for running postgres database .

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍 

Please can you add some docs to the README, on how to use this during development

**Comment by sital002** [comment]
> Please can you add some docs to the README, on how to use this during development

I have updated the documentation

File: README.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ All contributions welcome, not just code.
...
 6. Start the local development server with `npm run dev`.
 
+### Run Postgres using docker 
+
+#### Requirement 
+- Must have docker installed 
+```bash
+docker-compose up -d
+``` 
+
```
Comment (eddiejaoude) [inline README.md:55]:
```suggestion
### Postgres using Docker Compose

Note: you must have docker installed 

Run the docker command `docker-compose up -d`

```

File: README.md [hunk 2]
```diff
@@ -45,6 +45,12 @@ All contributions welcome, not just code.
 5. Seed the database by inserting example data with `npm run db:dev:seed` (if you need to reset the database, use `npm run db:dev:reset`).
 6. Start the local development server with `npm run dev`.
 
+### Postgres using Docker Compose
+
+Note: you must have docker installed 
+
+Run the docker command `docker-compose up -d`
```
Comment (Pradumnasaraf) [inline README.md:52]:
```suggestion
Run the docker command `docker compose up -d`
```

File: docker-compose.yaml
Comment (Pradumnasaraf) [inline docker-compose.yaml:1]: Change file name to `compose.yaml` from `docker-compose.yaml`

**Review by Pradumnasaraf — CHANGES_REQUESTED** [review]
Hey,

Thanks for your contribution! This command was used in Compose V1, which is deprecated. Also, with the new version, the file naming convention has been updated.

For reference, please check out these links:
- [Docker Compose Up Command](https://docs.docker.com/reference/cli/docker/compose/up)
- [Docker Compose File](https://docs.docker.com/compose/compose-application-model)

Let me know if you have issues.

Comments: **Hold merge**

**Review by Pradumnasaraf — APPROVED** [review]
Thank you for the changes. Looks good to merge!

**Comment by eddiejaoude** [comment]
Thank you both! Great collab 💪

**Pull request merged**